controller/src/comms.cpp

```diff
@@ -188,10 +215,76 @@ void comms_handler() {
     }
 }
 
-static void cmd_responseSend(uint8_t cmd, char *packet, uint8_t len) {
-    serialIO_send(msgType::rAck, (enum dataID) cmd, packet, len);
+
+void comms_sendResetState() {
+    char resetData[12];
+    uint32_t time;
+    char *version;
+
+    // NOTE : Given the bootloader clears MCUSR, it's not possible to determine what reset the CPU without modifying the bootloader
+
+    version = version_getVersion();
+
+    time = millis();
+
+    resetData[0] = (time >> 24) & 0xFF;
+    resetData[1] = (time >> 16) & 0xFF;
+    resetData[2] = (time >> 8) & 0xFF;
+    resetData[3] = time & 0xFF;
+
+    // TODO Is using memcpy faster than this?
```

Comment:
gcc will optimize this to memcpy if it thinks it's faster -- and it would transform memcpy to this loop if it thinks *that* would be faster.  I'd write it as a memcpy personally, but up to you.